### DISCUSSION TITLE
open.cc is looking for a file named ABBA.bin_param.bin_params, is the file extension a typo?

### DISCUSSION THREAD

**Discussion by tsamdin** [RosettaCommons/rosetta#159] [Rosetta] [answered]
Here's a section of the error log from Rosetta:

protocols.rosetta_scripts.RosettaScriptsParser: Defined mover named "fdes1" of type FastDesign
protocols.simple_moves.bin_transitions.PerturbByBins: Set bin params file name to "param/ABBA.bin_param".
basic.io.database: [ WARNING ] Unable to locate database file protocol_data/generalizedKIC/bin_params/param/ABBA.bin_param.bin_params
Error: [ ERROR ] ERROR: Exception caught by JobDistributor while trying to get pose from job 'output_Mut/no_loop_input_split_chain2_0001'
Error: [ ERROR ] Treating failure as bad input; canceling similar jobs

[ ERROR ]: Caught exception:

File: src/basic/database/open.cc:73
Database file open failed for: "protocol_data/generalizedKIC/bin_params/param/ABBA.bin_param.bin_params"

The files in main/database/protocol_data/generalizedKIC/bin_params include the following:

ABBA.bin_params
ABEGO.bin_params
PRO_DPRO.bin_params

I tried copying ABBA.bin_params and changing the file extension to ".bin_param.bin_params" but got the same error. 

For context I'm trying to use the PerturbByBins mover in my xml script. 

I'm not sure why basic.io.database is searching for "ABBA.bin_param.bin_params" as I used the bin_params_file="" variable in the PerturbByBins mover, and it seems like Rosetta can find that file without issue.

**Comment by tsamdin** [comment]
Removing ".bin_param" from the bin_params_file"" variable in the PerturbByBins mover seems to make it work. Should I assume that for all other movers that call on a ".bin_param" file I should leave off the file extension when I give the file path?

**Comment by roccomoretti** [comment, marked as answer]
Rosetta expects that the file being used for the `bin_params_file` option of PerturbByBins has an extension of ".bin_params" -- note the 's'. As a convenience, if the filename passed to `bin_params_file` does not have that extension, it will add it for you. It will first look for the file in the local directory, and if the file doesn't exist, it will attempt to locate it in the `protocol_data/generalizedKIC/bin_params/` database directory.

If you're attempting to use a custom local file, you would need to name it "param/ABBA.bin_params" (note 's') in order for it to work with PerturbByBins. Then you can pass `bin_params_file="param/ABBA.bin_params"` or `bin_params_file="param/ABBA"` to PerturbByBins.

If you're interested in one of the database files, you'll need to use just `bin_params_file="ABBA.bin_params"` or `bin_params_file="ABBA"` (the last is the default.) 
 
When you use the extension of ".bin_param" (note no 's'), it messes things up. Since it doesn't end with ".bin_params", Rosetta adds the extension for you, resulting in the "param/ABBA.bin_param.bin_params" -- this doesn't exist in the local directory, so it tries to find it in the database directory instead, which results in the observed error.

This behavior will be the same for any other mover which uses the BinTransitionCalculator (so likely anything else which takes a ".bin_params" file.)
